src/main/java/model/process/analysis/operations/comparisons/LagSequentialAnalysis.java

```diff
@@ -1,21 +1,233 @@
 
 /**
- * This class will determine a relation between events.
- * Relation will be shown by chronologically sorting events.
+ * This class will determine a relation between events. Relation will be shown
+ * by chronologically sorting events.
  * 
  * @author Louis Gosschalk 26-05-2015
  */
 public class LagSequentialAnalysis {
-	
+
+	private DataTable tableA;
+	private DataTable tableB;
+	private DataTable result;
+
+	private DataDescriber<DateTimeValue> dateA;
+	private DataDescriber<DateTimeValue> dateB;
+
+	private int positionA;
+	private int positionB;
+
+	private DateTimeValue compareA;
+	private DateTimeValue compareB;
+
+	private DataTableBuilder tableC;
+
+	private List<String> order;
+
 	/**
-	 * This class will put the events (tables) in one table, sorted chronologically.
+	 * This class constructs the LSA.
+	 * 
 	 * @param eventA
+	 *            The first event
+	 * @param dateA
+	 *            The column of the dateValues (first event)
 	 * @param eventB
+	 *            The second event
+	 * @param dateB
+	 *            The column of the dateValues (second event)
+	 * @return
+	 */
+	public LagSequentialAnalysis(Event eventA,
+			DataDescriber<DateTimeValue> dateA, Event eventB,
+			DataDescriber<DateTimeValue> dateB) {
+		tableA = checkTable(eventA.create());
+		tableB = checkTable(eventB.create());
+		this.dateA = dateA;
+		this.dateB = dateB;
+		this.tableC = new DataTableBuilder();
+
+		if (tableA.getRowCount() == 0 || tableB.getRowCount() == 0) {
+			throw new InputMismatchException("Empty event input.");
+		}
+
+		tableA = sortTable(tableA);
+		tableB = sortTable(tableB);
+
+		positionA = 0;
+		positionB = 0;
+
+		order = new ArrayList<String>();
+
+		compareA = dateA.resolve(tableA.getRow(positionA));
+		compareB = dateB.resolve(tableB.getRow(positionB));
+
+		chronoAdd();
+		tableC.setName("LSA");
+		result = tableC.build();
+	}
+
+	/**
+	 * Sort the table chronologically.
+	 * 
+	 * @param table
+	 *            the table to sort
+	 * @return a new table, sorted with the very same content
+	 */
+	public DataTable sortTable(DataTable table) {
+		DataTableBuilder res = new DataTableBuilder();
+		res.setName(table.getName());
+
+		DataDescriber<DateTimeValue> date = null;
+		if (table.getName().equals(tableA.getName())) {
+			date = dateA;
+		} else {
+			date = dateB;
+		}
+
+		List<Calendar> cal = new ArrayList<Calendar>();
+		DataRow row = null;
+		for (int i = 0; i < table.getRowCount(); i++) {
+			row = table.getRow(i);
+			cal.add(date.resolve(row).getValue());
+		}
+		Collections.sort(cal);
+
+		for (Calendar x : cal) {
+			for (int i = 0; i < table.getRowCount(); i++) {
+				row = table.getRow(i);
+				if (date.resolve(row).getValue().equals(x)) {
+					res.addRow(row);
+				}
+			}
+		}
+		return res.build();
+	}
+
+	/**
+	 * This class will put the events (rows) in one table, sorted
+	 * chronologically.
+	 */
+	public void chronoAdd() {
+		while (positionA < tableA.getRowCount()
```

Comment:
dat kan ook na de while loop en dat is een stuk gebruikelijker. zie ook hiervoor de comparison-example branch